CMakeLists.txt

```diff
@@ -280,12 +280,22 @@ add_custom_command(TARGET foedag POST_BUILD
           ${CMAKE_CURRENT_SOURCE_DIR}/third_party/tcl8.6.12/library/init.tcl
           ${CMAKE_CURRENT_BINARY_DIR}/lib/tcl8.6/init.tcl)
 
-execute_process(COMMAND git config --global --add safe.directory ${CMAKE_CURRENT_SOURCE_DIR}
-  COMMAND git rev-parse --short HEAD
-  COMMAND tr -d '\n'
-  OUTPUT_VARIABLE GIT_HASH
-  ERROR_VARIABLE GIT_HASH_ERR
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+if(WIN32)
+    execute_process(COMMAND git config --global --add safe.directory ${CMAKE_CURRENT_SOURCE_DIR}
+      COMMAND git rev-parse --short HEAD
+      COMMAND dos2unix
```

Comment:
Why do we need this? Git should ideally handle the line endings by depending on the OS.

Reply:
Hi @ravic-rs , in our local setup without this we are getting empty git hash value in the FOEDAG_GIT_HASH define, which results to a compilation error. Also we were testing the build from Raptor level as well.

Reply:
This is interesting, will come back to this later.